lib/Raven/Client.php

```diff
@@ -231,76 +110,67 @@ class Client
     /**
      * @var bool
      */
-    protected $_shutdown_function_has_been_set;
+    protected $_shutdown_function_has_been_set = false;
+
+    /**
+     * @var Configuration The client configuration
+     */
+    protected $config;
 
-    public function __construct($options_or_dsn = null, $options = [])
+    /**
+     * Constructor.
+     *
+     * @param Configuration $config The client configuration
+     */
+    public function __construct(Configuration $config)
```

Comment:
Why do you think that his approach is harder? 

Have you seen how much code he was able to delete thanks to this new approach? 
Did you see the `ClientBuilder` class that will help in easily instantiating a new client?